src/magic.cpp

```diff
@@ -647,6 +656,24 @@ bool spell::can_cast( const player &p ) const
             return p.get_power_level() >= units::from_kilojoule( energy_cost( p ) );
         case fatigue_energy:
             return p.get_fatigue() < EXHAUSTED;
+        case item_energy: {
+            const itype_id id = energy_item();
+            const inventory& player_inv = p.inv;
+            std::vector<const item*> a_filter = player_inv.items_with( [id](const item& it ) {
+                return it.typeId() == id;
+                });
+            if ( !a_filter.empty() ) {
+                if ( a_filter[ 0 ]->count_by_charges() ) {
+                    return player_inv.has_charges(id, energy_cost ( p ) );
```

Comment:
```C++
                if( item::count_by_charges( energy_item() ) ) {
                    return p.inv.has_charges( energy_item(), energy_cost( p ) );
                } else {
                    return p.inv.has_amount( energy_item(), energy_cost( p ) );
                }
```

And something similar can be used to consume the items.

Reply:
I could not find a function for depleting a specific number of charges from a specific item type, if one exists please point it out.